### PR TITLE
Juan3

### PR DESCRIPTION
**Pull request by jdcaicedo251**

### CODE REVIEW

File: activitysim/abm/models/initialize_from_usim.py [hunk 1]
```diff
@@ -130,46 +183,9 @@ def colleges(blocks):
...
-    null_blocks = pd.concat(result_list)
-    
-    # Concatenate newly assigned blocks to the main blocks table 
-    blocks_df = blocks_df.dropna()
-    blocks_df = pd.concat([blocks_df, null_blocks], axis = 0)
-    
-    return blocks_df.index_right
-
+    blocks_df = blocks.to_frame(columns = ['x', 'y'])
+    h3_gpd =  zones.to_frame(columns = ['geometry', 'area'])
```
Comment (mxndrwgrdnr) [inline activitysim/abm/models/initialize_from_usim.py:191]: Even though `zones.to_frame()` might return a GeoDataFrame object type, it does not persist the crs attribute, so you should explicitly set it before passing it to `assign_taz()`

File: activitysim/abm/models/initialize_from_usim.py [hunk 2]
```diff
@@ -202,47 +218,9 @@ def RESACRE(blocks):
...
-    null_school = pd.concat(result_list)
-
-    # Concatenate newly assigned blocks to the main blocks table 
-    school_gdf = school_gdf.dropna()
-    school_all = pd.concat([school_gdf, null_school], axis = 0)
-    school_all.set_index('ncessch', inplace = True)
-    return school_all.index_right
-
+    h3_gpd =  zones.to_frame(columns = ['geometry', 'area'])
+    school_gpd = orca.get_table('schools').to_frame(columns = ['x', 'y'])
```
Comment (mxndrwgrdnr) [inline activitysim/abm/models/initialize_from_usim.py:226]: same comment as above

File: activitysim/abm/models/initialize_from_usim.py [hunk 3]
```diff
@@ -294,26 +272,11 @@ def part_time_enrollment():
...
-    # Spatial join 
-    colleges_df = gpd.sjoin(colleges_df, h3_gpd, how = 'left', op = 'intersects')
-
-    #Drop duplicates and keep the one with the smallest H3 area
-    colleges_df = colleges_df.sort_values('area')
-    colleges_df.drop_duplicates(subset = ['x', 'y'], keep = 'first', inplace = True) 
-    
-    return colleges_df.index_right
-
+    h3_gpd =  zones.to_frame(columns = ['geometry', 'area'])
```
Comment (mxndrwgrdnr) [inline activitysim/abm/models/initialize_from_usim.py:282]: same as above

File: activitysim/abm/models/initialize_from_usim.py [hunk 4]
```diff
@@ -660,10 +623,17 @@ def COLLPTE(colleges, zones):
...
+    
+    mpo = mpo_taz.to_frame(columns = ['geometry','area_type','ACRES'])
+    h3_gpd =  zones.to_frame(columns = ['geometry', 'area'])
+    
+    join = gpd.sjoin(h3_gpd, mpo, how = 'left',op='intersects')
+    join.area_type.fillna(5, inplace = True) #Fill non-matched areas with 5 (Rural areas)
+    join = join.groupby(['TAZ', 'area_type'])['ACRES'].sum().reset_index()
+    join = join.sort_values(['TAZ', 'ACRES'], ascending = True)
+    s = join.groupby('TAZ')['area_type'].last()
+    return s
```
Comment (mxndrwgrdnr) [inline activitysim/abm/models/initialize_from_usim.py:640]: this looks great. just curious as to how you mapped the MPO's 5 area type categories to Activitysim's 6. Is that documented anywhere?

**Review by mxndrwgrdnr — CHANGES_REQUESTED** [review]
This all looks good Juan. Once you fix the crs issue we can merge.

File: activitysim/abm/models/initialize_from_usim.py (hunk 1 above)
Comment (jdcaicedo251) [reply activitysim/abm/models/initialize_from_usim.py:191]: great. I'll fixed the issue inside the `assign_taz()`.

File: activitysim/abm/models/initialize_from_usim.py (hunk 4 above)
Comment (jdcaicedo251) [reply activitysim/abm/models/initialize_from_usim.py:640]: mapping is in area_type_impute.py

**Pull request merged**